app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala

```diff
@@ -265,12 +312,32 @@ object SharesDisposalCYAController {
         )
         .withMarginBottom(Margin.Fixed60Bottom),
       refresh = None,
-      buttonText = parameters.mode.fold(
-        normal = "site.saveAndContinue",
-        check = "site.continue"
-      ),
-      onSubmit = routes.SharesDisposalCYAController
-        .onSubmit(parameters.srn, parameters.shareIndex, parameters.disposalIndex, parameters.mode)
+      buttonText =
+        parameters.mode.fold(normal = "site.saveAndContinue", check = "site.continue", viewOnly = "site.continue"),
+      onSubmit = controllers.nonsipp.sharesdisposal.routes.SharesDisposalCYAController
+        .onSubmit(parameters.srn, parameters.shareIndex, parameters.disposalIndex, parameters.mode),
+      optViewOnlyDetails = if (mode == ViewOnlyMode) {
+        Some(
+          ViewOnlyDetailsViewModel(
+            updated = viewOnlyUpdated,
+            link = None,
+            submittedText = Some(Message("")),
+            title = "sharesDisposal.viewOnly.title",
+            heading = Message("sharesDisposal.viewOnly.heading", parameters.companyName),
+            buttonText = "site.continue",
```

Comment:
should this be site.return.to.tasklist ?

Reply:
Here also same @kberec